frontend/src/features/public-form/components/FormFields/FormFields.tsx

```diff
@@ -27,18 +35,18 @@ export const FormFields = ({
   colorTheme,
   onSubmit,
 }: FormFieldsProps): JSX.Element => {
-  // TODO: Inject default values if field is also prefilled.
   const augmentedFormFields = useMemo(
-    () => formFields.map(augmentWithMyInfo),
+    () => formFields.map(augmentWithMyInfo).map(augmentWithMyInfoPrefill),
     [formFields],
   )
 
   const defaultFormValues = useMemo(() => {
     return augmentedFormFields.reduce<FormFieldValues>((acc, field) => {
-      if (field.fieldValue !== undefined) {
-        acc[field._id] = field.fieldValue
+      if (isMyInfoPrefilledFormField(field)) {
+        acc[field._id] = extractPrefilledValue(field)
```

Comment:
i think this should be reverted, previous code was correctly setting fieldValue (if it exists from the data the server returned)

edit: scrolled down and saw what `isMyInfoPrefilledFormField` actually does. maybe rename it haha. Good bug fix on the `extractPrefilledValue` function!